nvflare/private/fed_json_config.py

```diff
@@ -166,21 +183,27 @@ def finalize_config(self, config_ctx: ConfigContext):
 
         data_filter_table = {}
         for c in self.task_data_filter_chains:
-            if not isinstance(c, FilterChain):
-                raise TypeError("chain must be FilterChain but got {}".format(type(c)))
-            for t in c.tasks:
-                if t in data_filter_table:
-                    raise ConfigError("multiple data filter chains defined for task {}".format(t))
-                data_filter_table[t] = c.filters
+            self._build_filter_table(c, data_filter_table)
         self.data_filter_table = data_filter_table
 
         result_filter_table = {}
         for c in self.task_result_filter_chains:
-            if not isinstance(c, FilterChain):
-                raise TypeError("chain must be FilterChain but got {}".format(type(c)))
-            for t in c.tasks:
-                if t in result_filter_table:
-                    raise ConfigError("multiple data filter chains defined for task {}".format(t))
-                result_filter_table[t] = c.filters
-
+            self._build_filter_table(c, result_filter_table)
         self.result_filter_table = result_filter_table
+
+    def _build_filter_table(self, c, data_filter_table):
+        direction = c.direction.lower()
+        if not FilterChain.validate_direction(direction):
+            raise TypeError("Filter chain direction {} is not supported.".format(direction))
+        if not isinstance(c, FilterChain):
+            raise TypeError("chain must be FilterChain but got {}".format(type(c)))
+        for t in c.tasks:
+            if direction == FilterKey.INOUT:
+                directions = [FilterKey.IN, FilterKey.OUT]
+            else:
+                directions = [direction]
+            for item in directions:
+                task_key = t + FilterKey.DELIMITER + item
+                if task_key in data_filter_table:
+                    raise ConfigError("multiple data filter chains defined for task {}".format(task_key))
+                data_filter_table[task_key] = c.filters
```

Comment:
suggest change "task_key" to "task_filter_key" or something?
Since it contains both task name AND the direction